src/gsy_e/models/market/two_sided.py

```diff
@@ -344,7 +344,10 @@ def match_recommendations(
                     # re-raise exception to be handled by the external matcher
                     raise invalid_bop_exception
                 continue
-            original_bid_rate = market_bid.original_price / market_bid.energy
+            original_bid_rate = recommended_pair.bid_energy_rate - market_bid.accumulated_grid_fees
+            market_bid.update_price(
+                recommended_pair.bid_energy_rate * recommended_pair.selected_energy)
+            market_bid.original_price = original_bid_rate * recommended_pair.selected_energy
```

Comment:
If the matching algorithm has used the requirements' price value, we should update the bid's generic price because latter in the code we frequently read this value